src/Database/ODBC/SQLServer.hs

```diff
@@ -512,7 +532,12 @@ renderFractional x = trim (printf "%.7f" (realToFrac x :: Double) :: String)
                  s' -> s')
 
 renderTimeZone :: TimeZone -> String
-renderTimeZone tzone = if tzone == utc then "Z" else timeZoneOffsetString tzone
+renderTimeZone (TimeZone 0 _ _) = "Z"
+renderTimeZone (TimeZone t _ _) | t < 0 = '-' : renderTimeZone' (negate t)
+renderTimeZone (TimeZone t _ _) = '+' : renderTimeZone' t
+
+renderTimeZone' :: Int -> String
+renderTimeZone' t = printf "%02d:%02d" (t `div` 60) (t `mod` 60)
```

Comment:
This string-based rendering may not have the best performance, but as long as it's correct I don't mind merging.

Reply:
I chose to start with the simplest thing I could get to work. In general, I'm no performance optimisation expert, but I'm open to suggestions for better ways to do this.

As a general observation, I noticed that the _time_ library in version 1.9 added a module called `Data.Time.Format.ISO8601`, which, I speculate could replace much of the date and time formatting code - not only for `ZonedTime`/`Datetimeoffset`, but also possibly for some of the other time-related types.

Whether it's faster I don't know, but it might have the potential to simplify some of the code...